nautobot/utilities/filters.py

```diff
@@ -423,9 +428,58 @@ class SearchFilter(MappedPredicatesFilterMixin, django_filters.CharFilter):
 #
 
 
-class BaseFilterSet(django_filters.FilterSet):
+class BaseFilterSetFilterForm(forms.Form):
+    """Base/default filter form class for BaseFilterSet."""
+
+    def is_valid(self):
+        """Extend forms.Form.is_valid() to treat any unknown form data entries as a potential validation error."""
+        result = super().is_valid()
+        for extra_key in set(self.data.keys()).difference(self.cleaned_data.keys()):
```

Comment:
Another minor nit since casting a dict to a set/list/tuple does so on the keys:

```suggestion
        for extra_key in set(self.data).difference(self.cleaned_data):
```

Reply:
I think I'd prefer to keep this one as-is just to be more explicit about what is going on, if that's okay by you.